kyo-data/shared/src/main/scala/kyo/Show.scala

```diff
@@ -0,0 +1,93 @@
+            def show(value: A): String =
+                val builder = java.lang.StringBuilder()
+                builder.append(label)
+                builder.append("(")
+                val valIter                       = value.asInstanceOf[Product].productIterator
+                val showIter: Iterator[Show[Any]] = shows.productIterator.asInstanceOf
+                if valIter.hasNext then
+                    builder.append(showIter.next().show(valIter.next()))
+                    ()
+                while valIter.hasNext do
+                    builder.append(",")
+                    builder.append(showIter.next().show(valIter.next()))
+                    ()
+                end while
+                builder.append(")")
+                builder.toString()
+            end show
+        end new
+    end sumShow
+
+    inline given [A](using mir: scala.deriving.Mirror.Of[A]): Show[A] = inline mir match
+        case sumMir: scala.deriving.Mirror.SumOf[?] =>
+            val shows = summonAll[Tuple.Map[sumMir.MirroredElemTypes, Show]]
+            new Show[A]:
+                def show(value: A): String =
+                    val caseIndex               = sumMir.ordinal(value)
+                    val showInstance: Show[Any] = shows.productElement(caseIndex).asInstanceOf
+                    showInstance.show(value)
+                end show
+            end new
+        case singMir: scala.deriving.Mirror.Singleton =>
+            val label: String = constValue[singMir.MirroredLabel]
+            new Show[A]:
+                def show(value: A): String = label
+        case prodMir: scala.deriving.Mirror.ProductOf[?] => inline erasedValue[A] match
+                case _: Tuple =>
+                    inline erasedValue[prodMir.MirroredElemTypes] match
+                        case _: EmptyTuple =>
+                            new Show[A]:
+                                def show(value: A): String = "()"
+                        case _ =>
+                            sumShow[A, prodMir.type]("", prodMir)
+                case _ =>
+                    val label: String = constValue[prodMir.MirroredLabel]
+                    inline erasedValue[prodMir.MirroredElemTypes] match
+                        case _: EmptyTuple =>
+                            new Show[A]:
+                                def show(value: A): String = label + "()"
+                        case _ =>
+                            sumShow[A, prodMir.type](label, prodMir)
+                    end match
+        // case _ =>
+        //     error("unknown mirror type")
+
+end Show
+
+sealed trait Shown:
+    private[kyo] def shownValue: String
+
+object Shown:
+    given [A](using sh: Show[A]): Conversion[A, Shown] with
+        def apply(a: A): Shown =
+            new Shown:
+                private[kyo] def shownValue: String = sh.show(a)
+    end given
+end Shown
+
+extension (sc: StringContext)
+    def k(args: Shown*): String = sc.s(args.map(_.shownValue)*)
```

Comment:
I don't really understand the purpose of `Text`, is it meant to be a Kyo-wide replacement for `String`? Will `Console.printLine` and `Log.info` accepts `Text` instead of `String`?

If so, one thing we could is replace `Show` with `AsText`, and instead of using the `Conversion[A: Show, Shown]` that I'm using for interpolation, we could use `Conversion[A: AsText, Text]`, and have the interpolator generate a single combined `Text`, as you suggest.

Any thoughts?

Reply:
> I don't really understand the purpose of Text, is it meant to be a Kyo-wide replacement for String? Will Console.printLine and Log.info accepts Text instead of String?

Yes. It currently only addresses an important performance issue with string operations that require array copying but we'll likely add new string-related features to it and implement more advanced optimizations. We should update `Console` to take `Text`. Note that `Text` is defined as a super type of `String` so it's possible to pass plain strings to methods that take `Text`. @hearnadam could you share your thoughts on this?

> If so, one thing we could is replace Show with AsText, and instead of using the Conversion[A: Show, Shown] that I'm using for interpolation, we could use Conversion[A: AsText, Text], and have the interpolator generate a single combined Text, as you suggest.

It might be better to restrict the conversion to a specific type like `Shown`. Wouldn't the conversion trigger for any type that has a `show` evidence, essentially adding the methods in `Text` to all showable types?

Reply:
Yes good point -- that's why I made the method on `Shown` private.

I think, then, I will:
- Make `Show` as `AsText`, and have it convert to `Text` instead of `String` (this way we won't conflict with `Show` for anyone using cats)
- Update `k` to generate `Text` instead of `String`
- Update `Log` and `Console` to use `Text`

(Let me know if this doesn't make sense)

Reply:
Sounds good. I think the interpolator could `text` instead of `k`? If you prefer, we can make the Console change separately

Reply:
Is `txt` okay? `text` can conflict with other methods with that name.